src/emqx_misc.erl

```diff
@@ -466,6 +468,63 @@ maybe_mute_rpc_log(Node) ->
             ok
     end.
 
+is_sensitive_key(token) -> true;
+is_sensitive_key("token") -> true;
+is_sensitive_key(<<"token">>) -> true;
+is_sensitive_key(password) -> true;
+is_sensitive_key("password") -> true;
+is_sensitive_key(<<"password">>) -> true;
+is_sensitive_key(secret) -> true;
+is_sensitive_key("secret") -> true;
+is_sensitive_key(<<"secret">>) -> true;
+is_sensitive_key(passcode) -> true;
+is_sensitive_key("passcode") -> true;
+is_sensitive_key(<<"passcode">>) -> true;
+is_sensitive_key(passphrase) -> true;
+is_sensitive_key("passphrase") -> true;
+is_sensitive_key(<<"passphrase">>) -> true;
+is_sensitive_key(_) -> false.
+
+redact(L) when is_list(L) ->
+    lists:map(fun redact/1, L);
+redact(M) when is_map(M) ->
+    maps:map(fun(K, V) ->
+                     redact(K, V)
+             end, M);
+redact(T) when is_tuple(T) ->
+    Redact = case is_nested_tuple(T) of
+                 {true, Elements} ->
+                     redact(Elements);
+                 {false, [Head | Tail]} ->
+                     [Head | redact(Head, Tail)]
+             end,
+    erlang:list_to_tuple(Redact);
```

Comment:
```suggestion
redact(T) when is_tuple(T) ->
    list_to_tuple(redact(tuple_to_list(T)).
```

Reply:
I think we still need to iterate over the tuple,
e.g.
```
{{shutdown,{failed_to_start_child,worker_sup,
                                  {shutdown,{failed_to_start_child,{worker,1},
                                                                   {connection_error,econnrefused}}}}},
 {child,undefined,
        {pool_sup,'backend_redis:resource:723930'},
        {ecpool_pool_sup,start_link,
                         ['backend_redis:resource:723930',emqx_backend_redis_actions,
                          [{sentinel,[]},
                           {pool_size,1},
                           {database,0},
                           {password,"12312321321312321"},
                           {auto_reconnect,2},
                           {host,"127.0.0.1"},
                           {port,6379},
                           {options,[]}]]},
        transient,false,infinity,supervisor,
        [ecpool_pool_sup]}}
```